### DISCUSSION TITLE
Log probability generation

### DISCUSSION THREAD

**Discussion by AngelaZhang3913** [ggml-org/llama.cpp#9500] [Q&A] [answered]
I am using curl requests to display the log probabilities of the next tokens. Below is part of the raw response I get from the curl request. Here, the content is 4, but the token with the highest probability (of 0.419) is 2, not 4. Can someone explain why this is happening? Shouldn't the content that is outputted be the same as the token with the highest log probability?

`[{""content"":""4"",""probs"":[{""tok_str"":""2"",""prob"":0.41857030987739563},{""tok_str"":""1"",""prob"":0.2892381250858307},{""tok_str"":""4"",""prob"":0.2227334976196289},{""tok_str"":""3"",""prob"":0.023897262290120125},{""tok_str"":""I"",""prob"":0.017724353820085526},{""tok_str"":""a"",""prob"":0.015108339488506317},{""tok_str"":""Nothing"",""prob"":0.012728091329336166},{""tok_str"":""'"",""prob"":0.0}]}]}`

**Comment by ggerganov** [comment, marked as answer]
> Shouldn't the content that is outputted be the same as the token with the highest log probability?

No - this would be the case only with the "greedy" sampling strategy. For example, if you set the temperature in your request to <= 0.0, it will always choose the highest-probability token. But by default, the temperature is positive so there is proportional chance to select some of the lower-probability tokens.
